java/arcs/android/host/prod/ProdArcHostService.kt

```diff
@@ -30,31 +31,40 @@ import kotlinx.coroutines.ExperimentalCoroutinesApi
  */
 @ExperimentalCoroutinesApi
 @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-open class ProdArcHostService : ArcHostService() {
+abstract class ProdArcHostService : ArcHostService() {
     @ExperimentalCoroutinesApi
     class ProdAndroidHost(
         context: Context,
         lifecycle: Lifecycle,
+        coroutineContext: CoroutineContext,
+        arcSerializationCoroutineContext: CoroutineContext,
         schedulerProvider: SchedulerProvider,
         vararg particles: ParticleRegistration
     ) : AndroidHost(
         context = context,
         lifecycle = lifecycle,
+        coroutineContext = coroutineContext,
+        arcSerializationContext = arcSerializationCoroutineContext,
         schedulerProvider = schedulerProvider,
         particles = *particles
     ), ProdHost
 
+    abstract val coroutineContext: CoroutineContext
+    abstract val arcSerializationCoroutineContext: CoroutineContext
```

Comment:
I think these could use some especially detailed KDoc to explain what they're used for and to help the developer to decide which contexts to pass.

Reply:
Done. I improved the naming of `coroutineContext` to reflect its current usage, as well.

This is all due for a refactor as well, it's on my todo list once the HandleManagerProvider work is done (the structure there will make it a lot easier to plumb coroutine contexts through).